test/e2e/magic_catalog_test.go

```diff
@@ -0,0 +1,40 @@
+package e2e
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/operator-framework/operator-lifecycle-manager/test/e2e/ctx"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var _ = Describe("MagicCatalog", func() {
+	var (
+		generatedNamespace corev1.Namespace
+	)
+
+	BeforeEach(func() {
+		generatedNamespace = SetupGeneratedTestNamespace(genName("magic-catalog-e2e-"))
+	})
+
+	AfterEach(func() {
+		TeardownNamespace(generatedNamespace.GetName())
+	})
+
+	It("Deploys and Undeploys a File-based Catalog", func() {
+		// create dependencies
+		const catalogName = "test"
+		namespace := generatedNamespace.GetName()
+		kubeClient := ctx.Ctx().Client()
+		provider, err := NewFileBasedFiledBasedCatalogProvider("../test/e2e/testdata/fbc_catalog.json")
```

Comment:
Personally, I'd rather see the FBC JSON literal in the test case itself than in a golden file, but this works for now.

Reply:
tbh - I'm not a huge fan of big string literals, especially when we can't nicely format them due to code formatting or indentation, etc. For small stuff sure, for bigger files it's start to get awkward for _me_. But this is a taste thing. I've tried to create an abstraction through this provider interface where we can plug-in different fbc sources (file, string, etc.). When I have a little more time, I'd like to have a way to create the fbcs programatically. Might be even more readable that way...idk...